core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -47,12 +48,12 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   /** Create a SparkConf that loads defaults from system properties and the classpath */
   def this() = this(true)
 
-  private[spark] val settings = new TrieMap[String, String]()
+  private[spark] val settings = new ConcurrentHashMap[String, String]()
```

Comment:
while you are at it, define the return type explicitly here (java.util.Map?) since it is public type